liberty/LibertyReader.cc

```diff
@@ -3065,6 +3069,23 @@ LibertyReader::visitClockGatingIntegratedCell(LibertyAttr *attr)
   }
 }
 
+void
+LibertyReader::visitCellFootprint(LibertyAttr *attr)
+{
+  if (cell_) {
+    const char *footprint = getAttrString(attr);
+    LibertyCellFootprintIndex footprint_index;
+    if (footprint_index_map_.find(footprint) != footprint_index_map_.end()) {
+      footprint_index = footprint_index_map_.at(footprint);
```

Comment:
You are looking it up twice.  If you capture the iterator from find you can avoid needing to call at.